### PR TITLE
Tests and bug fixes for core

### PR DESCRIPTION
**Pull request by Pessimistress**

### CODE REVIEW

File: modules/core/src/index.js [hunk 1]
```diff
@@ -36,15 +36,7 @@ export {
   isWritableStream
 } from './javascript-utils/is-type';
 
-export {
-  isArrayBuffer,
-  isBlob,
```
Comment (Pessimistress) [inline modules/core/src/index.js]: These are not used anywhere

File: modules/core/src/javascript-utils/binary-utils.js [hunk 2]
```diff
@@ -1,19 +1,14 @@
-/* global FileReader, Blob, ArrayBuffer, Buffer, TextEncoder */
+/* global ArrayBuffer, TextEncoder */
 import assert from '../utils/assert';
-
-export const isArrayBuffer = x => x && x instanceof ArrayBuffer;
-export const isBlob = x => x && typeof Blob !== 'undefined' && x instanceof Blob;
-export const isBuffer = x => x && x instanceof Buffer;
```
Comment (Pessimistress) [inline modules/core/src/javascript-utils/binary-utils.js]: This is breaking the browser if Buffer is not polyfilled

File: modules/core/src/javascript-utils/memory-copy-utils.js [hunk 3]
```diff
@@ -1,57 +1,3 @@
-export function padTo4Bytes(byteLength) {
```
Comment (Pessimistress) [inline modules/core/src/javascript-utils/memory-copy-utils.js]: Duplicate - Modules are importing these from loader-utils

File: .travis.yml [hunk 4]
```diff
@@ -12,4 +12,4 @@ install:
 script:
   - npm run test ci
 after_success:
-  - cat coverage/lcov.info | coveralls
+  - if [[ `node -v` == *"v12"* ]]; then (cat coverage/lcov.info | coveralls); fi
```
Comment (ibgreen) [inline .travis.yml]: Great, I was about to try something like this to tackle the flaky coverage. Does it work?

File: modules/core/src/javascript-utils/binary-utils.js [hunk 5]
```diff
@@ -1,19 +1,14 @@
...
+import {toArrayBuffer as bufferToArrayBuffer} from '../node/utils/to-array-buffer.node';
 
 export function toArrayBuffer(data) {
-  if (isArrayBuffer(data)) {
-    return data;
+  if (bufferToArrayBuffer) {
+    data = bufferToArrayBuffer(data);
   }
 
-  // TODO - per docs we should just be able to call buffer.buffer, but there are issues
```
Comment (ibgreen) [inline modules/core/src/javascript-utils/binary-utils.js]: Maybe keep this comment on line 6?

File: .travis.yml (hunk 4 above)
Comment (Pessimistress) [reply .travis.yml]: Yes.

**Pull request merged**